### ISSUE TITLE
I want to publish the application in a single file, but I have encountered a problem

### ISSUE THREAD

**Issue by fldx** [webview/webview_csharp#28]
### Demo address 

https://github.com/fldx/DesktopApp

### Publish command
```ps1
dotnet publish -c Release -p:EnableCompressionInSingleFile=true -p:PublishTrimmed=true
```

### Running results
![image](https://user-images.githubusercontent.com/11643460/217184303-3b634a25-8b95-4e52-8465-a52f729f0921.png)

### Compiled log
```
PS D:\VSProjects\DesktopApp> dotnet publish -c Release -p:PublishSingleFile=true -p:PublishTrimmed=true
MSBuild version 17.4.0+18d5aef85 for .NET
  正在确定要还原的项目…
  所有项目均是最新的，无法还原。
  DesktopApp -> D:\VSProjects\DesktopApp\bin\Release\net6.0\win-x64\DesktopApp.dll
  正在优化程序集以调整大小。此过程可能需要一段时间。
ILLink : Trim analysis warning IL2026: System.Data.DataSet.System.Xml.Serialization.IXmlSerializable.GetSchema(): Using member 'System.Data.DataSet.WriteXmlSchema(DataSet, XmlWriter)' which has 'RequiresUnreferencedCodeAttribute' can break funct
ionality when trimming application code. DataSet.GetSchema uses TypeDescriptor and XmlSerialization underneath which are not trimming safe. Members from serialized types may be trimmed if not referenced directly. [D:\VSProjects\DesktopApp\Deskto
pApp.csproj]
ILLink : Trim analysis warning IL2026: System.Data.DataSet.System.Xml.Serialization.IXmlSerializable.ReadXml(XmlReader): Using member 'System.Data.DataSet.ReadXmlSerializableInternal(XmlReader)' which has 'RequiresUnreferencedCodeAttribute' can
break functionality when trimming application code. DataSet.ReadXml uses XmlSerialization underneath which is not trimming safe. Members from serialized types may be trimmed if not referenced directly. [D:\VSProjects\DesktopApp\DesktopApp.csproj
]
ILLink : Trim analysis warning IL2026: System.Data.DataSet.System.Xml.Serialization.IXmlSerializable.WriteXml(XmlWriter): Using member 'System.Data.DataSet.WriteXmlInternal(XmlWriter)' which has 'RequiresUnreferencedCodeAttribute' can break func
tionality when trimming application code. DataSet.WriteXml uses XmlSerialization underneath which is not trimming safe. Members from serialized types may be trimmed if not referenced directly. [D:\VSProjects\DesktopApp\DesktopApp.csproj]
ILLink : Trim analysis warning IL2026: System.Data.DataTable.System.Xml.Serialization.IXmlSerializable.GetSchema(): Using member 'System.Data.DataTable.GetXmlSchema()' which has 'RequiresUnreferencedCodeAttribute' can break functionality when tr
imming application code. DataTable.GetSchema uses TypeDescriptor and XmlSerialization underneath which are not trimming safe. Members from serialized types may be trimmed if not referenced directly. [D:\VSProjects\DesktopApp\DesktopApp.csproj]
ILLink : Trim analysis warning IL2026: System.Data.DataTable.System.Xml.Serialization.IXmlSerializable.ReadXml(XmlReader): Using member 'System.Data.DataTable.ReadXmlSerializableInternal(XmlReader)' which has 'RequiresUnreferencedCodeAttribute'
can break functionality when trimming application code. DataTable.ReadXml uses XmlSerialization underneath which is not trimming safe. Members from serialized types may be trimmed if not referenced directly. [D:\VSProjects\DesktopApp\DesktopApp.
csproj]
ILLink : Trim analysis warning IL2026: System.Data.DataTable.System.Xml.Serialization.IXmlSerializable.WriteXml(XmlWriter): Using member 'System.Data.DataTable.WriteXmlInternal(XmlWriter)' which has 'RequiresUnreferencedCodeAttribute' can break
functionality when trimming application code. DataTable.WriteXml uses XmlSerialization underneath which is not trimming safe. Members from serialized types may be trimmed if not referenced directly. [D:\VSProjects\DesktopApp\DesktopApp.csproj]
C:\Users\Admin\.nuget\packages\microsoft.aspnetcore.connections.abstractions\2.2.0\lib\netstandard2.0\Microsoft.AspNetCore.Connections.Abstractions.dll : warning IL2104: Assembly 'Microsoft.AspNetCore.Connections.Abstractions' produced trim warn
ings. For more information see https://aka.ms/dotnet-illink/libraries [D:\VSProjects\DesktopApp\DesktopApp.csproj]
C:\Users\Admin\.nuget\packages\microsoft.aspnetcore.diagnostics\2.2.0\lib\netstandard2.0\Microsoft.AspNetCore.Diagnostics.dll : warning IL2104: Assembly 'Microsoft.AspNetCore.Diagnostics' produced trim warnings. For more information see https://
aka.ms/dotnet-illink/libraries [D:\VSProjects\DesktopApp\DesktopApp.csproj]
C:\Users\Admin\.nuget\packages\microsoft.aspnetcore.hosting\2.2.0\lib\netstandard2.0\Microsoft.AspNetCore.Hosting.dll : warning IL2104: Assembly 'Microsoft.AspNetCore.Hosting' produced trim warnings. For more information see https://aka.ms/dotne
t-illink/libraries [D:\VSProjects\DesktopApp\DesktopApp.csproj]
C:\Users\Admin\.nuget\packages\microsoft.aspnetcore.http.abstractions\2.2.0\lib\netstandard2.0\Microsoft.AspNetCore.Http.Abstractions.dll : warning IL2104: Assembly 'Microsoft.AspNetCore.Http.Abstractions' produced trim warnings. For more inform
ation see https://aka.ms/dotnet-illink/libraries [D:\VSProjects\DesktopApp\DesktopApp.csproj]
C:\Users\Admin\.nuget\packages\microsoft.aspnetcore.http.extensions\2.2.0\lib\netstandard2.0\Microsoft.AspNetCore.Http.Extensions.dll : warning IL2104: Assembly 'Microsoft.AspNetCore.Http.Extensions' produced trim warnings. For more information
see https://aka.ms/dotnet-illink/libraries [D:\VSProjects\DesktopApp\DesktopApp.csproj]
C:\Users\Admin\.nuget\packages\microsoft.aspnetcore.routing.abstractions\2.2.0\lib\netstandard2.0\Microsoft.AspNetCore.Routing.Abstractions.dll : warning IL2104: Assembly 'Microsoft.AspNetCore.Routing.Abstractions' produced trim warnings. For mo
re information see https://aka.ms/dotnet-illink/libraries [D:\VSProjects\DesktopApp\DesktopApp.csproj]
C:\Users\Admin\.nuget\packages\microsoft.aspnetcore.routing\2.2.0\lib\netcoreapp2.2\Microsoft.AspNetCore.Routing.dll : warning IL2104: Assembly 'Microsoft.AspNetCore.Routing' produced trim warnings. For more information see https://aka.ms/dotnet
-illink/libraries [D:\VSProjects\DesktopApp\DesktopApp.csproj]
C:\Users\Admin\.nuget\packages\microsoft.extensions.configuration.binder\2.2.0\lib\netstandard2.0\Microsoft.Extensions.Configuration.Binder.dll : warning IL2104: Assembly 'Microsoft.Extensions.Configuration.Binder' produced trim warnings. For mo
re information see https://aka.ms/dotnet-illink/libraries [D:\VSProjects\DesktopApp\DesktopApp.csproj]
C:\Users\Admin\.nuget\packages\microsoft.extensions.dependencyinjection.abstractions\2.2.0\lib\netstandard2.0\Microsoft.Extensions.DependencyInjection.Abstractions.dll : warning IL2104: Assembly 'Microsoft.Extensions.DependencyInjection.Abstract
ions' produced trim warnings. For more information see https://aka.ms/dotnet-illink/libraries [D:\VSProjects\DesktopApp\DesktopApp.csproj]
C:\Users\Admin\.nuget\packages\microsoft.extensions.dependencyinjection\2.2.0\lib\netcoreapp2.0\Microsoft.Extensions.DependencyInjection.dll : warning IL2104: Assembly 'Microsoft.Extensions.DependencyInjection' produced trim warnings. For more i
nformation see https://aka.ms/dotnet-illink/libraries [D:\VSProjects\DesktopApp\DesktopApp.csproj]
C:\Users\Admin\.nuget\packages\microsoft.extensions.logging.eventsource\2.2.0\lib\netstandard2.0\Microsoft.Extensions.Logging.EventSource.dll : warning IL2104: Assembly 'Microsoft.Extensions.Logging.EventSource' produced trim warnings. For more
information see https://aka.ms/dotnet-illink/libraries [D:\VSProjects\DesktopApp\DesktopApp.csproj]
C:\Users\Admin\.nuget\packages\microsoft.extensions.logging\2.2.0\lib\netstandard2.0\Microsoft.Extensions.Logging.dll : warning IL2104: Assembly 'Microsoft.Extensions.Logging' produced trim warnings. For more information see https://aka.ms/dotne
t-illink/libraries [D:\VSProjects\DesktopApp\DesktopApp.csproj]
C:\Users\Admin\.nuget\packages\microsoft.extensions.options\2.2.0\lib\netstandard2.0\Microsoft.Extensions.Options.dll : warning IL2104: Assembly 'Microsoft.Extensions.Options' produced trim warnings. For more information see https://aka.ms/dotne
t-illink/libraries [D:\VSProjects\DesktopApp\DesktopApp.csproj]
C:\Users\Admin\.nuget\packages\newtonsoft.json\11.0.2\lib\netstandard2.0\Newtonsoft.Json.dll : warning IL2104: Assembly 'Newtonsoft.Json' produced trim warnings. For more information see https://aka.ms/dotnet-illink/libraries [D:\VSProjects\Desk
topApp\DesktopApp.csproj]
C:\Users\Admin\.nuget\packages\sharpwebview\0.7.0\lib\netstandard2.1\SharpWebview.dll : warning IL2104: Assembly 'SharpWebview' produced trim warnings. For more information see https://aka.ms/dotnet-illink/libraries [D:\VSProjects\DesktopApp\Des
ktopApp.csproj]
  DesktopApp -> D:\VSProjects\DesktopApp\bin\Release\net6.0\win-x64\publish\
```

**Comment by fldx** [comment]
在非托管代码调用时返回null
https://learn.microsoft.com/en-us/dotnet/api/system.reflection.assembly.getentryassembly?redirectedfrom=MSDN&view=net-7.0#System_Reflection_Assembly_GetEntryAssembly
https://github.com/webview/webview_csharp/blob/d830a9bec7ba8fe192cb79466ab96f7ebd76a763/src/SharpWebview/Content/HostedContent.cs#L51

**Comment by geaz** [comment]
Fixed by #29
